packages/core/app/mirage/routes/report.js

```diff
@@ -3,22 +3,68 @@ import moment from 'moment';
 import RESPONSE_CODES from '../enums/response-codes';
 const TIMESTAMP_FORMAT = 'YYYY-MM-DD HH:mm:ss';
 
+/**
+ * @method getFilterParams
+ * @param {String} queryFilter
+ * @returns {Array} Filter parameters
+ * @description Parse filter parameters in the form of "(title==*H*,request==*Revenue*)";author==*ramvish*
+ * to a list of all the OR parameters, ie., [H, Revenue]
+ */
+const getFilterParams = function(queryFilter) {
+  return queryFilter
+    .split(';')[0]
+    .replace(/\(/g, '')
+    .replace(/\)/g, '')
+    .replace(/\*/g, '')
+    .split(',')
+    .map(el => el.split('==')[1]);
+};
+
+/**
+ * @method getQueryAuthor
+ * @param {String} queryFilter
+ * @returns Author
+ * @description Parse filter parameters in the form of "(title==*H*,request==*Revenue*)";author==*ramvish*
+ * to get the author, ie., ramvish
+ */
+const getQueryAuthor = function(queryFilter) {
+  if (queryFilter.includes('author')) {
+    return queryFilter
+      .split(';')[1]
+      .replace(/\*/g, '')
+      .split('==')[1];
+  }
+  return null;
+};
+
 export default function() {
   /**
    * reports/ - GET endpoint to fetch many reports
    */
   this.get('/reports', function({ reports }, request) {
+    let reportsObject;
     let idFilter = request.queryParams['filter[reports.id]'];
+    let queryFilter = request.queryParams['filter[reports]'];
 
     // Allow filtering
     if (idFilter) {
       let ids = idFilter.split(',');
-      reports = reports.find(ids);
+      reportsObject = reports.find(ids);
+    } else if (queryFilter) {
+      let filterParameters = getFilterParams(queryFilter);
+      let author = getQueryAuthor(queryFilter);
+      reportsObject = reports.all().filter(function(report) {
+        const matchesFilterParameter = filterParameters.every(filterParameter =>
+          JSON.stringify(report).match(new RegExp(filterParameter, 'i'))
```

Comment:
Would be better to Stringify the report outside of the `every` callback since the result will be the same with each call.
Would be best to keep track of the property the filter value is being applied on and stringifying that property in the report to search for. 

If a filter contains `title==*H*` then we should do `JSON.stringify(report.title)`